File: CodeBase/Nesi.Main/NESI.DTO/ViewModels/Page/TimeSheet/UpdateTimeSheetShop.cs
using System.ComponentModel.DataAnnotations;

namespace NESI.DTO.ViewModels.Page.TimeSheet
{
	public class UpdateTimeSheetShop : UpdateTimeSheetBase
	{
		[Required]
		public int MemberTime_ID { get; set; }
		[Range(minimum: 0, maximum: 5)]
		public int Rating { get; set; }
		[Required]
		public double NumberOfHours { get; set; }
		[Required]
		public int MemberTime_WoComment_ID { get; set; }
		[Required]
		public string MemberTime_WoComment { get; set; }
        public int? Internal_Project_Id { get; set; }

    }
}

File: CodeBase/Nesi.Main/angular/src/app/models/pages/timesheet/updateTimesheetShop.ts
import { UpdateTimeSheetBase } from './UpdateTimesheetBase';
export interface UpdateTimesheetShop  extends UpdateTimeSheetBase {
  memberTime_ID: number;
  rating: number;
  numberOfHours: number;
  memberTime_WoComment_ID: number;
  memberTime_WoComment: string;
  internal_project_id : number;
}
